superset-frontend/src/SqlLab/components/TabbedSqlEditors/index.jsx

```diff
@@ -260,8 +256,31 @@ class TabbedSqlEditors extends React.PureComponent {
       : t(
           '-- Note: Unless you save your query, these tabs will NOT persist if you clear your cookies or change browsers.\n\n',
         );
+
+    let newTitle;
+
+    if (this.props.queryEditors.length > 0) {
+      const untitledQueryNumbers = this.props.queryEditors
+        .filter(x => x.title.includes('Untitled Query '))
+        .map(x => x.title.replace('Untitled Query ', ''))
+        .filter(x => !Number.isNaN(Number(x)));
```

Comment:
maybe let's find all tab titles that match a regex of `^Untitled Query (\d+)$`, and then we can pull the number our of the group that matches there. Then we also shouldn't need to do the `isNaN` check.

Reply:
That seems like a much better way of only getting the query tabs with the title "Untitled Query #". Thank you for the suggestion, I had added this change to the latest commit :)